crates/iceberg/src/scan.rs

```diff
@@ -207,6 +268,34 @@ impl FileScanTask {
     }
 }
 
+/// Evaluates manifest files to see if their partition values comply with a filter predicate
+pub struct PartitionEvaluator {
+    bound_predicate: BoundPredicate,
+    partition_spec: PartitionSpecRef,
+}
+
+impl PartitionEvaluator {
+    pub(crate) fn new(partition_spec: PartitionSpecRef, bound_predicate: BoundPredicate) -> Self {
+        PartitionEvaluator {
+            bound_predicate,
+            partition_spec,
+        }
+    }
+
+    pub(crate) fn filter_manifest(&self, _manifest: &Manifest) -> bool {
+        // TODO: filter manifests using partition summaries
```

Comment:
How about leaving a `todo!()` here to avoid we forgot this TODO and leak this behavior to end users?